e2e/playwright/projects.spec.ts

```diff
@@ -605,3 +605,114 @@ test(
     await electronApp.close()
   }
 )
+
+test(
+  'You can change the root projects directory and nothing is lost',
+  { tag: '@electron' },
+  async ({ browserName }, testInfo) => {
+    const { electronApp, page } = await setupElectron({
+      testInfo,
+      folderSetupFn: async (dir) => {
+        await Promise.all([
+          fsp.mkdir(`${dir}/router-template-slate`, { recursive: true }),
+          fsp.mkdir(`${dir}/bracket`, { recursive: true }),
+        ])
+        await Promise.all([
+          fsp.copyFile(
+            'src/wasm-lib/tests/executor/inputs/router-template-slate.kcl',
+            `${dir}/router-template-slate/main.kcl`
+          ),
+          fsp.copyFile(
+            'src/wasm-lib/tests/executor/inputs/focusrite_scarlett_mounting_braket.kcl',
+            `${dir}/bracket/main.kcl`
+          ),
+        ])
+      },
+    })
+    await page.setViewportSize({ width: 1200, height: 500 })
+
+    page.on('console', console.log)
+
+    await page.waitForTimeout(1_000)
+
+    // we'll grab this from the settings on screen before we switch
+    let originalProjectDirName: string
+    const newProjectDirName = testInfo.outputPath(
+      'electron-test-projects-dir-2'
+    )
+    if (fs.existsSync(newProjectDirName)) {
+      await fsp.rm(newProjectDirName, { recursive: true })
+    }
+
+    await test.step('We can change the root project directory', async () => {
+      // expect to see the project directory settings link
+      await expect(
+        page.getByTestId('project-directory-settings-link')
+      ).toBeVisible()
+
+      await page.waitForTimeout(100)
+
+      page.getByTestId('project-directory-settings-link').click()
```

Comment:
I think all these yellow squiggles need to be awaited.

Run `yarn eslint e2e`. There's probably already some warnings on `main`, but I'm hoping we can avoid adding new issues.

<img width="750" alt="Screenshot 2024-08-16 at 11 02 48 AM" src="https://github.com/user-attachments/assets/fe26f234-efbd-424f-9274-c2face41a346">

Reply:
Great catch! Yeah not awaiting clicks can definitely lead to racy behavior

Reply:
Thanks, this explains my races!